sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConsumer.cs

```diff
@@ -61,6 +61,19 @@ internal class AmqpConsumer : TransportConsumer
         ///
         private string PartitionId { get; }
 
+        /// <summary>
+        ///   The current position for the consumer, updated as events are received from the
+        ///   partition.
+        /// </summary>
+        ///
+        /// <remarks>
+        ///   When creating or recovering the associated AMQP link, this value is used
+        ///   to set the position.  It is intended to primarily support recreating links
+        ///   transparently to callers, allowing progress in the stream to be remembered.
+        /// </remarks>
+        ///
+        private EventPosition CurrentEventPosition { get; set; }
```

Comment:
I didn't really get the point of this property. You're storing the `CurrentEventPosition`, but overwrites it before opening the consumer link. And it's `private`, so I don't see how it should be used.

Reply:
The link can die and need to be reopened.   If it does, it should use the last event that was received as its starting point.  Otherwise, any time the link closes because it was idle or because an error force closed it, the factory would restart from the specified initial position.

Reply:
```csharp
ReceiveLink = new FaultTolerantAmqpObject<ReceivingAmqpLink>(timeout =>
                ConnectionScope.OpenConsumerLinkAsync(
                    consumerGroup,
                    partitionId,
                    CurrentEventPosition,
                    timeout,
                    prefetchCount ?? DefaultPrefetchCount,
                    ownerLevel,
                    trackLastEnqueuedEventProperties,
                    CancellationToken.None),
                link => link.SafeClose());
```

Notice that fault tolerant object takes a factory to create the link?   That is responsible for understanding whether the link is alive and in good health before handing it back or whether a new instance needs to be created.   It's transparent to us as the caller and managed by the blackbox AMQP library.